src/mongodb/scores.controller.ts

```diff
@@ -699,137 +1572,1444 @@ export class ScoresController {
+      complexityLevel = ["C1"]
+    } else if (currentLevel === 'm3') {
+      contentLevel = 'L3';
+      complexityLevel = ["C1", "C2"];
+    } else if (currentLevel === 'm4') {
+      contentLevel = 'L4';
+      complexityLevel = ["C1", "C2", "C3"]
+    }
+
+    let getGetTargetCharArr = getGetTarget.filter((getGetTargetEle, index) => {
+      if (gettargetlimit !== 0 && index >= gettargetlimit) {
+        return false;
+      }
+      return true;
+    }).map(charData => {
+      return charData.character
+    });
+
+    const url = process.env.ALL_CONTENT_SERVICE_API;
+
+    const textData = {
+      "tokenArr": getGetTargetCharArr,
+      "language": language || "ta",
+      "contentType": "Paragraph",
+      "limit": contentlimit || 5,
+      "tags": tags,
+      "cLevel": contentLevel,
+      "complexityLevel": complexityLevel
+    };
+
+    const newContent = await lastValueFrom(
+      this.httpService.post(url, JSON.stringify(textData), {
+        headers: {
+          'Content-Type': 'application/json',
+        }
+      }).pipe(
+        map((resp) => resp.data)
+      )
+    );
+
+    let contentArr;
+    let contentForTokenArr;
+
+    if (newContent.data.hasOwnProperty('wordsArr')) {
+      contentArr = newContent.data.wordsArr;
+    } else {
+      contentArr = [];
+    }
+
+    if (newContent.data.hasOwnProperty('contentForToken')) {
+      contentForTokenArr = newContent.data.contentForToken;
+    } else {
+      contentForTokenArr = [];
+    }
+
+
+    return { content: contentArr, contentForToken: contentForTokenArr, getTargetChar: getGetTargetCharArr, currentLevel: currentLevel };
   }
 
-  @Get('/GetConfidentSet/session/:sessionId')
-  @ApiOperation({ summary: 'Get ConfidentSet character API by session id' })
+  @ApiParam({
+    name: "sessionId",
+    example: "27519278861697549531193"
+  })
+  @Get('/getMilestoneProgress/session/:sessionId')
+  @ApiOperation({ summary: 'Get Milstone Progress by session id' })
   @ApiResponse({
     status: 200,
-    description: 'Success response with Get ConfidentSet Word character and score for session id',
+    description: 'Success response with Get Targets character and score for user id',
     schema: {
       properties: {
-        character: { type: 'string' },
-        score: { type: 'number', format: 'float' },
+        currentLevel: { type: 'string' },
+        targetsCount: { type: 'number', format: 'float' },
+        validationsCount: { type: 'number', format: 'float' },
+
       }
     },
   })
-  GetConfidentSetBysession(@Param('sessionId') id: string) {
-    return this.scoresService.getConfidentSetBySession(id);
+  async getMilestoneForSession(@Param('sessionId') id: string) {
+    let targets = await this.scoresService.getTargetsBySession(id);
+    let validations = await this.scoresService.getAssessmentRecords(id);
+
+    let totalTargets = targets.length;
+    let totalValidation = validations.length;
+
+    let currentLevel = 'm1';
+
+    if (totalTargets >= 30) {
```

Comment:
Is this still relevant?  Also all these numbers should be configurable, not hardcoded